src/Core/src/Platform/Windows/MauiWinUIWindow.cs

```diff
@@ -25,5 +36,39 @@ protected virtual void OnVisibilityChanged(object sender, UI.Xaml.WindowVisibili
 		{
 			MauiWinUIApplication.Current.Services?.InvokeLifecycleEvents<WindowsLifecycle.OnVisibilityChanged>(del => del(this, args));
 		}
+
+		public event EventHandler<WindowsNativeMessageEventArgs> NativeMessage;
+
+		#region Native Window
+		IntPtr _hwnd = IntPtr.Zero;
+		delegate IntPtr WinProc(IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam);
+		WinProc? newWndProc = null;
+		IntPtr oldWndProc = IntPtr.Zero;
+
+		[DllImport("user32")]
+		static extern IntPtr SetWindowLongPtr(IntPtr hWnd, int nIndex, WinProc newProc);
+		[DllImport("user32.dll")]
+		static extern IntPtr CallWindowProc(IntPtr lpPrevWndFunc, IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam);
+
+		void SubClassingWin32()
+		{
+			//Get the Window's HWND
+			_hwnd = this.As<IWindowNative>().WindowHandle;
+			if (_hwnd == IntPtr.Zero)
+				throw new NullReferenceException("The Window Handle is null.");
+
+			newWndProc = new WinProc(NewWindowProc);
+			oldWndProc = SetWindowLongPtr(_hwnd, /* GWL_WNDPROC */ -4, newWndProc);
+		}
+
+		IntPtr NewWindowProc(IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam)
+		{
+			var args = new WindowsNativeMessageEventArgs(hWnd, msg, wParam, lParam);
+
+			NativeMessage?.Invoke(this, args);
```

Comment:
It might be worth rewriting these lines to :
```cs
_NativeMessage?.Invoke(this, new WindowsNativeMessageEventArgs(hWnd, msg, wParam, lParam));
```
That way you don't allocate the event arguments unless there's a listener.